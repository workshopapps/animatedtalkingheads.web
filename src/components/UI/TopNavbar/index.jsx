import { Link } from "react-router-dom";
import styles from "./styles.module.css";
import headerLogo from "./../../../assets/icons/header-logo.png";
import menuIcon from "./../../../assets/icons/menu-icon.png";

const links = [
  { name: "Home", link: "/"},
  { name: "About Us", link: "/about"},
  { name: "Use cases", link: "/use-cases"},
  { name: "Contact", link: "/contact"},
  { name: "How it work", link: "/how-it-works"},
]

const TopNavbar = () => {
  return (
    <div className={styles.nav}>
      <div className={`h-full flex justify-between items-center px-3 lg:px-20`}>
        <div className="flex">
          <Link to="/"> <img src={headerLogo} alt="home"/></Link>
        </div>
        <div className={`${styles.links} hidden md:flex justify-between basis-3/5`}>
          { links.map( (link, index) => <Link className="block" key={index} to={link.link}> {link.name} </Link> ) }
        </div>
        <div className={`${styles.create} hidden md:flex`}>
          <Link to="/podcast/upload" className="btn px-10 py-3">
            Create Video
          </Link>
        </div>
        <button className="md:hidden">
          <img src={menuIcon} alt="home"/>
        </button>
      </div>
    </div>
  );
};

export default TopNavbar;
