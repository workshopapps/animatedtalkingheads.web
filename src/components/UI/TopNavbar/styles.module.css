.nav {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background: #fbfbfb;
  border-bottom: 1px solid rgba(68, 68, 68, 0.1);
  height: 4em;
  color: #292d32;
  z-index: 10000;
}

.nav .links {
  max-width: 420px;
}

.nav .links a {
  font-size: 1em;
  line-height: 20px;
  color: #292d32;
}

.nav .links a:hover {
  color: #2158d2;
}

.nav .links .active {
  color: #2158d2;
}

.navDummy {
  height: 4em;
}

.menu {
  background: rgba(251, 251, 251, 0.8);
  backdrop-filter: blur(7.5px);
  position: fixed;
  top: 4em;
  left: 0;
  width: 100%;
  height: calc(100% - 4em);
  z-index: 100000;
}

.menu .links {
  font-weight: 600;
  font-size: 16px;
  line-height: 20px;
  color: #606368;
  margin-bottom: 1.25em;
}

.menu a {
  font-weight: 600;
  font-size: 16px;
  line-height: 20px;
  color: #606368;
  margin-bottom: 1.25em;
}

.menu .line {
  border: 0.5px solid #afb0b3;
}

.menu .links2 {
  margin-top: 14px;
}

.menu .links2 a {
  font-weight: 600;
  font-size: 14px;
  line-height: 18px;
  color: #606368;
  margin-top: 14px;
}

@media screen and (min-width: 480px) {
  .nav {
    height: 5.5em;
  }

  .nav .links {
  }

  .navDummy {
    height: 5.5em;
  }
}

.createVideo {
  position: absolute;
  width: 240px;
  height: 58px;
  background: #2563eb;
  border-radius: 8px;
  color: #fefefe !important;
  line-height: 30px;
  display: flex;
  justify-content: center;
  align-items: center;
}

@media screen and (max-width: 756px) {
  a:hover {
    color: #292d32;
  }
}
