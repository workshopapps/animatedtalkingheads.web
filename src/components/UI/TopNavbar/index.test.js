import React from 'react';
import { render, screen } from '@testing-library/react';
import TopNavbar from './';
import { BrowserRouter } from 'react-router-dom';

test('Renders TopNavbar', () => {
    render(<TopNavbar />, {wrapper: BrowserRouter})


});
