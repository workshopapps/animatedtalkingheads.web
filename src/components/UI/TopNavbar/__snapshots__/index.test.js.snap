// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Test should render and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nav"
  >
    <div
      class="h-full flex justify-between items-center px-3 lg:px-20"
    >
      <div
        class="flex"
      >
        <a
          href="/"
        >
           
          <img
            alt="home"
            src="header-logo.png"
          />
        </a>
      </div>
      <div
        class="links hidden md:flex justify-between basis-3/5"
      >
        <a
          class="block"
          href="/"
        >
           Home 
        </a>
        <a
          class="block"
          href="/about"
        >
           About Us 
        </a>
        <a
          class="block"
          href="/use-cases"
        >
           Use cases 
        </a>
        <a
          class="block"
          href="/contact"
        >
           Contact 
        </a>
        <a
          class="block"
          href="/how-it-works"
        >
           How it work 
        </a>
        <a
          class="block"
          href="/avatars"
        >
           Avatars 
        </a>
      </div>
      <div
        class="hidden md:flex"
      >
        <a
          href="/customize-audio"
        >
          <button
            class="button false primary"
            type="button"
          >
            <p
              class="btn_text"
            >
              Create Video
            </p>
          </button>
        </a>
      </div>
      <button
        class="md:hidden hover:border-0"
      >
        <img
          alt="home"
          src="menu-icon.png"
        />
      </button>
    </div>
    <div
      class="menu hidden px-3 py-10 md:hidden"
    >
      <div
        class="links"
      >
        <a
          class="block"
          href="/"
        >
           Home 
        </a>
        <a
          class="block"
          href="/about"
        >
           About Us 
        </a>
        <a
          class="block"
          href="/use-cases"
        >
           Use cases 
        </a>
        <a
          class="block"
          href="/contact"
        >
           Contact 
        </a>
        <a
          class="block"
          href="/how-it-works"
        >
           How it work 
        </a>
        <a
          class="block"
          href="/avatars"
        >
           Avatars 
        </a>
      </div>
      <div
        class="line"
      >
         
      </div>
      <div
        class="links2"
      >
        <a
          class="block"
          href="/"
        >
           Support 
        </a>
        <a
          class="block"
          href="/settings"
        >
           Settings 
        </a>
        <a
          class="block"
          href="/community"
        >
           Community 
        </a>
      </div>
      <div
        class="line"
      >
         
      </div>
      <div
        class="flex w-full justify-center h-28 items-center"
      >
        <a
          class="createVideo"
          href="/customize-audio"
        >
          <button>
             Create Video 
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
